optional_plugins/varianter_yaml_to_mux/avocado_varianter_yaml_to_mux/__init__.py

```diff
@@ -234,10 +239,10 @@ def mapping_to_tree_loader(loader, node, looks_like_node=False):
         _value = []
         for key_node, value_node in node.value:
             # Allow only strings as dict keys
-            if key_node.tag.startswith('!'):    # reflect tags everywhere
+            if key_node.tag.startswith('!'):  # reflect tags everywhere
                 key = loader.construct_object(key_node)
             else:
-                key = loader.construct_python_str(key_node)
+                key = loader.construct_scalar(key_node)
```

Comment:
Took me a while to understand what if a scalar should be constructed here... according to 

https://yaml.org/spec/1.1/#scalar/information%20model

This seems correct.